Packs/Netskope/Integrations/NetskopeEventCollector/NetskopeEventCollector.py

```diff
@@ -0,0 +1,316 @@
+                                      headerTransform=string_to_table_header)
+
+    return CommandResults(outputs_prefix='Netskope.Event',
+                          outputs_key_field='_id',
+                          outputs=events,
+                          readable_output=readable_output,
+                          raw_response=events)
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:  # pragma: no cover
+    params = demisto.params()
+
+    url = params.get('url')
+    api_version = params.get('api_version')
+    token = demisto.params().get('credentials', {}).get('password')
+    base_url = urljoin(url, f'/api/{api_version}/')
+    verify_certificate = not demisto.params().get('insecure', False)
+    proxy = demisto.params().get('proxy', False)
+    first_fetch = params.get('first_fetch')
+    max_fetch = params.get('max_fetch')
+    vendor, product = params.get('vendor', 'netskope'), params.get('product', 'netskope')
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        client = Client(base_url, token, api_version, verify_certificate, proxy)
+
+        last_run = demisto.getLastRun()
+        if 'alert' not in last_run and 'application' not in last_run and 'audit' not in last_run \
+                and 'network' not in last_run:
+            last_run = arg_to_seconds_timestamp(first_fetch)
+            last_run = {
+                'alert': last_run,
+                'application': last_run,
+                'audit': last_run,
+                'network': last_run
+            }
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client, api_version, last_run)
+            return_results(result)
+
+        elif demisto.command() == 'netskope-get-events':
+            if api_version == 'v1':
+                return_results(v1_get_events_command(client, demisto.args(), last_run))
+            else:
+                return_results(v2_get_events_command(client, demisto.args(), last_run))
+        elif demisto.command() == 'fetch-events':
+            if api_version == 'v1':
+                events = client.get_events_request_v1(last_run, max_fetch)
+                alerts = client.v1_get_alerts_request(last_run, max_fetch)
+                if alerts:
+                    events.extend(alerts)
+                demisto.setLastRun(get_last_run(events, last_run))
+                demisto.debug(f'Setting the last_run to: {last_run}')
+                send_events_to_xsiam(events=events, vendor=vendor, product=product)
+            else:
+                events = client.get_events_request_v2(last_run, max_fetch)
+                demisto.setLastRun(get_last_run(events, last_run))
+                demisto.debug(f'Setting the last_run to: {last_run}')
+                send_events_to_xsiam(events=events, vendor=vendor, product=product)
```

Comment:
Let's split this into two functions. In main you will have:
```python
events = fetch_events_command(bla bla)
demisto.setLastRun(get_last_run(events, last_run))
demisto.debug(f'Setting the last_run to: {last_run}')
send_events_to_xsiam(events=events, vendor=vendor, product=product)
```
And in fetch_events_command you will have something like:
```python  
if api_version == 'v1':
       events = client.get_events_request_v1(last_run, max_fetch)
       alerts = client.v1_get_alerts_request(last_run, max_fetch)
       if alerts:
            events.extend(alerts)
else:
       events = client.get_events_request_v2(last_run, max_fetch)
```
This way you will not repeat code twice AND you will be able to call the fetch from the test module command.